avoidance/src/avoidance_node.cpp

```diff
@@ -21,21 +37,17 @@ AvoidanceNode::AvoidanceNode(const ros::NodeHandle& nh, const ros::NodeHandle& n
   statusloop_spinner_.reset(new ros::AsyncSpinner(1, &statusloop_queue_));
   statusloop_spinner_->start();
 
-  position_received_ = true;
-
-  timeout_termination_ = 15;
-  timeout_critical_ = 0.5;
-  timeout_startup_ = 5.0;
+  worker_ = std::thread(&AvoidanceNode::checkPx4Parameters, this);
```

Comment:
Just curious, is there are reason we are using threads instead of the `ros::AsyncSpinner` ?

Reply:
We want the timing to be different depending on whether the parameters are initialized yet or not.